### ISSUE TITLE
Edges are sometimes not deletable in v10 with the backspace key

### ISSUE THREAD

**Issue by AndrewRayCode** [wbkd/react-flow#2000]
### Describe the Bug

In v10, there's a new bug where an edge is not deletable when pressing backspace.

### Steps to Reproduce the Bug or Issue

Unknown. Something like:

1. On an initial render of a graph with existing edges, click on an edge and press backspace to delete it (this works)
2. Drag a new edge to a new connection. I use custom handles, I don't know if that's relevant.
3. Click on the new edge to select it and press backspace

Nothing happens on step 3.

I'm using controlled edge updates, if that's relevant. 

### Expected behavior

Edge is deleted

### Platform

- OS: Mac
- Browser: Chrome
- Version: 10.0.3

### Additional context

Confirmed by at least one other user https://discord.com/channels/771389069270712320/859774873500778517/956063307461177344

**Comment by moklick** [comment]
I can't reproduce this. Can you create a repo or codesandbox ?

**Comment by Sec-ant** [comment]
I cannot reproduce this but I can confirm I encountered this several times. Not only the backspace key is not working, but it appears that any keyboard event cannot be fired, e.g.: drag selection, multi-selection. It usually happens when I stop interacting with the page where RF runs on, and switch back to it later, but I'm not that sure this is the reason. I also doubt that maybe it's caused by the using of 3rd party packages, which may introduce confilicts when they also have their own key event handling mechanism. Just this afternoon, I ran into this problem again, and it seems that multi-selecting mode is always active, re-press the `multiSelectionKey` won't deactive the mode, and other keys like `Shift`, `Delete`/`Backspace` do not work, too.

What I would appreiciate is that someone can shed some light on how to debug this issue, like which variable should I watch, or other approaches to help me narrow down the problem.

**Comment by raarts** [comment]
I had the backspace key not working occasionally as well. Wasn't able to reproduce it at will.
Just upgraded from 10.0.1 to 10.0.6. Hasn't happened since, but that's not definitive yet. I'll update if I know more.

**Comment by moklick** [comment]
v10.0.6 resets the keys on tab change. I hope that it was the reason for this issue. Does the problem still occur @AndrewRayCode

**Comment by raarts** [comment]
I've just had it happen again (in 10.0.6). Whenever this happens, I can select/deselect nodes by clicking on them, but the Backspace does nothing. Also, Shift cannot select a group of nodes any longer, instead the Shift-key is ignored and panning happens instead.

This only seems to happen after I read a set of nodes/edges from the database, replacing the existing ones.

**Comment by iamfreund** [comment]
I have encountered a similar issue when a custom node contains an input field and the value is changed to a specific value. 

I am using `"react-flow-renderer": "10.0.7"`

In my particular scenario, the value I am using is "Functional". So if you change the value of the input field to that specific value and then select the left connection line and press the backspace key, the expected behavior of deleting the selected element does not work. If you press the backspace key a second time, deleting the selected element works as expected.

I have created a codesandbox example that should allow you to reproduce the behavior described above.

https://codesandbox.io/s/react-flow-edge-selection-bug-6l8yc4

Steps to reproduce:

1. Click in the input field, delete the current value and type the word "Functional".
2. Click somewhere on the area outside the node.
3. Select the left connection line and press the backspace key.

The element cannot be deleted immediately. The element can only be deleted after pressing the Backspace key again.

However, if you try the whole thing with another word, such as "Test", then the selected element can be deleted immediately.

Notes worth mentioning:

The behavior differs between versions `10.0.6` and `10.0.7`. In version `10.0.6` the selected element cannot be deleted at all with the backspace key.

This specific change seems to affect the behavior between the two versions: https://github.com/wbkd/react-flow/commit/b413c768dd8f93e1524adb327e42d41f334b8ad7
